### ISSUE TITLE
max-w-* malfunction?

### ISSUE THREAD

**Issue by flowtitude** [tailwindlabs/tailwindcss#15348]
**What version of Tailwind CSS are you using?**

Tailwind 4.0 Beta 6

**What build tool (or framework if it abstracts the build tool) are you using?**

WordPress

I'm not sure if it's a bug, but it used to work fine with the first betas, I don't know if it's since beta 4 maybe. I'm defining in @theme some flexible spaces with a terminology --spacing-4xs, spacing-3xs up to the spacing-4xl value, the problem is that max-w-4xl is taking the value of spacing-4xl in the latest betas, I think it would be more correct to take it from --container-4xl, or am I wrong, if this is correct how could I define semantic type spaces without interfering with max-w-*? thanks for the effort, this is looking great ;-)

**Comment by wongjn** [comment]
Could be due to https://github.com/tailwindlabs/tailwindcss/pull/15180 that was released in Beta 3.

**Comment by RobinMalfait** [comment]
Hey!

As @wongjn mentioned, we changed this behavior a little bit where we read from the spacing case by default for these utilities. If you were relying on values from the container, you can be explicit about it by using an arbitrary value. More info: https://github.com/tailwindlabs/tailwindcss/pull/15180

In your case, this means `max-w-4xl` would be reading from `--spacing-4xl`, you can still read from `--container-4xl` by using `max-w-(--container-4xl)`. E.g.: https://play.tailwindcss.com/oHMz7YC9AM

The main motivation here is to make it simpler where all these values use the spacing scale by default (unless overwritten by more specific values).

So long story short, this was a change made on purpose and not a bug, but you can use `max-w-(--container-4xl)` to be explicit.

Hope this helps!

**Comment by marcospgp** [comment]
the docs are still wrong on this today:

<img width="733" height="619" alt="Image" src="https://github.com/user-attachments/assets/4f3e1fd8-75f9-479f-8969-ef5f8da57521" />

and IMO it makes zero sense to use spacing vars for dimension styles (width, height, flex basis, etc)
